src/Microsoft.AspNetCore.Mvc.Core/Internal/MiddlewareFilterBuilder.cs

```diff
@@ -71,13 +71,20 @@ private RequestDelegate BuildPipeline(Type middlewarePipelineProviderType)
                 var resourceExecutionDelegate = feature.ResourceExecutionDelegate;
 
                 var resourceExecutedContext = await resourceExecutionDelegate();
+                if (resourceExecutedContext == null || resourceExecutedContext.ExceptionHandled)
```

Comment:
In what case **valid** is this null? If it's only possible in invalid cases then just let it crash.